src/Monolog/JsonSerializableDateTimeImmutable.php

```diff
@@ -0,0 +1,48 @@
+<?php declare(strict_types=1);
+
+/*
+ * This file is part of the Monolog package.
+ *
+ * (c) Jordi Boggiano <j.boggiano@seld.be>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Monolog;
+
+use DateTimeZone;
+
+/**
+ * Overrides default json encoding of date time objects
+ *
+ * @author Menno Holtkamp
+ * @author Jordi Boggiano <j.boggiano@seld.be>
+ */
+class JsonSerializableDateTimeImmutable extends \DateTimeImmutable implements \JsonSerializable
+{
+    private bool $useMicroseconds;
+
+    public function __construct(bool $useMicroseconds, ?DateTimeZone $timezone = null)
+    {
+        $this->useMicroseconds = $useMicroseconds;
+
+        // if you like to use a custom time to pass to Logger::addRecord directly,
+        // call modify() or setTimestamp() on this instance to change the date after creating it
```

Comment:
Doesn't immutable mean that you can't modify the instance?

I tested locally, and set

Reply:
> This class behaves the same as [DateTime](https://www.php.net/manual/en/class.datetime.php) except new objects are returned when modification methods such as [DateTime::modify()](https://www.php.net/manual/en/datetime.modify.php) are called.

- https://www.php.net/manual/en/class.datetimeimmutable.php

Reply:
You need to use the instance returned by setTimestamp and pass that to Logger::addRecord as custom datetime instance.